endpointmanager/pkg/chplquerier/productquerier.go

```diff
@@ -0,0 +1,313 @@
+			if err != nil {
+				return errors.Wrap(err, "updating health IT product object failed")
+			}
+			err = store.UpdateHealthITProduct(ctx, existingDbProd)
+			if err != nil {
+				return errors.Wrap(err, "updating health IT product to store failed")
+			}
+		}
+	}
+	return nil
+}
+
+// determines if a product needs to be udpated.
+//
+// if the two products are equal, do not update.
+// else if the new product has a more recent certification edition than the exisitng product, update.
+// else if the new product has a more recent certification date than the exisitng product, update.
+// else if the new product has more certification criteria than the existing product, update.
+//
+// throws errors if
+// - the certification edition is not a year
+// - the certification criteria list is the same length but not equal
+// - the two products are not equal but their differences don't fall into the categories noted above.
+func prodNeedsUpdate(existingDbProd *endpointmanager.HealthITProduct, newDbProd *endpointmanager.HealthITProduct) (bool, error) {
+	// check if the two are equal.
+	if existingDbProd.Equal(newDbProd) {
+		return false, nil
+	}
+
+	// begin by comparing certification editions.
+	// Assumes certification editions are years, which is the case as of 11/20/19.
+	existingCertEdition, err := strconv.Atoi(existingDbProd.CertificationEdition)
+	if err != nil {
+		return false, errors.Wrap(err, "unable to make certification edition into an integer - expect certification edition to be a year")
+	}
+	newCertEdition, err := strconv.Atoi(newDbProd.CertificationEdition)
+	if err != nil {
+		return false, errors.Wrap(err, "unable to make certification edition into an integer - expect certification edition to be a year")
+	}
+
+	// if new prod has more recent cert edition, should update.
+	if newCertEdition > existingCertEdition {
+		return true, nil
+	} else if newCertEdition < existingCertEdition {
+		return false, nil
+	}
+
+	// cert editions are the same. if new prod has more recent cert date, should update.
+	if existingDbProd.CertificationDate.Before(newDbProd.CertificationDate) {
+		return true, nil
+	} else if existingDbProd.CertificationDate.After(newDbProd.CertificationDate) {
+		return false, nil
+	}
+
+	// cert dates are the same. checking certification criteria lists. if new prod has more criteria, should update.
+	if len(existingDbProd.CertificationCriteria) < len(newDbProd.CertificationCriteria) {
+		return true, nil
+	} else if len(existingDbProd.CertificationCriteria) > len(newDbProd.CertificationCriteria) {
+		return false, nil
+	}
+
+	// certification criteria lists are the same lengths. unknown precedent for updates. return error.
+	if !cmp.Equal(existingDbProd.CertificationCriteria, newDbProd.CertificationCriteria) {
+		return false, errors.New("HealthITProducts certification edition and date are equal, but has same number but unequal certification criteria - unknown precendence for updates")
```

Comment:
This error message is kind of confusing to me. is it saying that the number of certification criteria are equal but they are different criteria?